ca/config.go

```diff
@@ -138,27 +139,43 @@ func (s *SecurityConfig) UpdateRootCA(cert, key []byte, certExpiry time.Duration
 	}
 
 	// the RootCA pool should validate against the TLS certificate in the credentials
-	if s.ClientTLSCreds != nil {
-		s.ClientTLSCreds.UpdateCAs(rootCA.Pool, nil)
+	clientConfig := s.ClientTLSCreds.Config()
+	updatedClientConfig, err := NewClientTLSConfig(clientConfig.Certificates, rootCA.Pool, ManagerRole)
```

Comment:
Wouldn't this reintroduce this issue: https://github.com/docker/swarmkit/pull/1983#discussion_r102872515 ?